provider/lib/utils.js

```diff
@@ -216,41 +273,47 @@ module.exports = function(
-                    var apikey = trigger.doc.apikey;
-                    var triggerIdentifier = that.getTriggerIdentifier(apikey, namespace, name);
-                    logger.info(method, 'Checking if trigger', triggerIdentifier, 'still exists');
-
-                    var host = 'https://' + routerHost +':'+ 443;
-                    var triggerURL = host + '/api/v1/namespaces/' + namespace + '/triggers/' + name;
-                    var auth = apikey.split(':');
-
-                    request({
-                        method: 'get',
-                        url: triggerURL,
-                        auth: {
-                            user: auth[0],
-                            pass: auth[1]
-                        }
-                    }, function(error, response) {
-                        //delete from database if trigger no longer exists (404)
-                        if (!error && response.statusCode === 404) {
-                            logger.info(method, 'trigger', triggerIdentifier, 'could not be found');
-                            that.deleteTriggerFromDB(triggerIdentifier);
-                        }
-                        else {
-                            that.createTrigger(trigger.doc)
-                            .then(triggerIdentifier => {
-                                logger.info(method, triggerIdentifier, 'created successfully');
-                            })
-                            .catch(err => {
-                                logger.error(method, err);
-                                that.deleteTriggerFromDB(triggerIdentifier);
-                            });
-                        }
-                    });
+                    if (!trigger.doc.status || trigger.doc.status.active === true) {
+                        //check if trigger still exists in whisk db
+                        var namespace = trigger.doc.namespace;
+                        var name = trigger.doc.name;
+                        var apikey = trigger.doc.apikey;
+                        var triggerIdentifier = that.getTriggerIdentifier(apikey, namespace, name);
+                        logger.info(method, 'Checking if trigger', triggerIdentifier, 'still exists');
+
+                        var host = 'https://' + routerHost + ':' + 443;
+                        var triggerURL = host + '/api/v1/namespaces/' + namespace + '/triggers/' + name;
+                        var auth = apikey.split(':');
+
+                        request({
+                            method: 'get',
+                            url: triggerURL,
+                            auth: {
+                                user: auth[0],
+                                pass: auth[1]
+                            }
+                        }, function (error, response) {
+                            //disable trigger in database if trigger no longer exists (404)
+                            if (!error && response.statusCode === 404) {
```

Comment:
yeah, good point.  I will update to use the new shouldDisableTrigger for all these checks now